### PR TITLE
Rating feature

### PR DESCRIPTION
**Pull request by kgeriiie**
Added rating feature implementations to orders. It will store your rating in Firestore and you can edit it anytime after you submit it.

### CODE REVIEW

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/main/presentation/MainComponent.kt [hunk 1]
```diff
@@ -84,6 +90,14 @@ class DefaultMainComponent(
         navigation.replaceCurrent(configuration)
     }
 
+    private fun submitRating(rating: Rating) {
```
Comment (ermanergoz) [inline shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/main/presentation/MainComponent.kt:93]: While testing, I noticed that if there are several canceled orders in the "view all" section, and I leave a comment, all those orders end up having the same rating and comment

Comment (kgeriiie) [reply shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/main/presentation/MainComponent.kt:93]: Yes you are right, ux mistake :D I will figure out how to deal with that, maybe I can tide it to that order, however, if somebody cancels an order they won't rate the order itself or us :D maybe I will remove the rating button in that case..

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/orders/presentation/OrderHistoryComponent.kt [hunk 2]
```diff
@@ -111,13 +111,11 @@ open class OrderHistoryComponent(
             is OrderHistoryEvents.FetchContents -> fetchOrderHistory()
             is OrderHistoryEvents.PrimaryButtonPressed -> when(event.order.status) {
                 OrderStatus.PENDING -> commonUiState.update { it.copy(confirmAction = OrderHistoryConfirmAction.Cancellation(event.order)) }
```
Comment (ermanergoz) [inline shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/orders/presentation/OrderHistoryComponent.kt:113]: While testing, I noticed that there is an issue preventing me from canceling an order on the Order History screen

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/orders/presentation/OrderHistoryComponent.kt [hunk 3]
```diff
@@ -111,13 +111,11 @@ open class OrderHistoryComponent(
...
-                OrderStatus.SHIPPED -> Unit // TODO: implement feature
-                OrderStatus.CANCELLED -> Unit // TODO: implement feature
+                else -> onComponentEvent(OrderHistoryEvents.ShowRatingDialog)
```
Comment (ermanergoz) [inline shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/orders/presentation/OrderHistoryComponent.kt:114]: Also the Rating Dialog doesn't show when I click on Rate us button but it works great on view all screen

**Review by ermanergoz — COMMENTED** [review]
Fancy!

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/main/presentation/MainComponent.kt (hunk 1 above)
Comment (kgeriiie) [reply shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/main/presentation/MainComponent.kt]: Modified it :) Now users can rate completed (shipped)-orders which makes much more sense than before :)

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/orders/presentation/OrderHistoryComponent.kt (hunk 2 above)
Comment (kgeriiie) [reply shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/orders/presentation/OrderHistoryComponent.kt:141]: Fixed :)

**Comment by kgeriiie** [comment]
@ermanergoz You can check it if you have time :)

**Review by ermanergoz — APPROVED** [review]

**Pull request merged**